native/cocos/renderer/core/PassUtils.cpp

```diff
@@ -157,7 +157,19 @@ const ccstd::unordered_map<gfx::Type, GFXTypeWriterCallback> type2writer = {
          a[idx + 1] = p->y;
      }},
     {gfx::Type::FLOAT3, [](float *a, const MaterialProperty &v, index_t idx) {
-         if (ccstd::holds_alternative<Vec3>(v)) {
+         if (ccstd::holds_alternative<float>(v)) {
+             // data may be crupted, prevent crash here
+             float f = ccstd::get<float>(v);
+             a[idx] = f;
+             a[idx + 1] = 0;
+             a[idx + 2] = 0;
+         } else if (ccstd::holds_alternative<Vec2>(v)) {
+             // data may be crupted, prevent crash here
+             const auto &vec3 = ccstd::get<Vec2>(v);
+             a[idx] = vec3.x;
+             a[idx + 1] = vec3.y;
+             a[idx + 2] = 0;
+         } else if (ccstd::holds_alternative<Vec3>(v)) {
```

Comment:
Most of time, its Vec3,so Vec3 should be checked first.

Reply:
fixed